addons/stock/models/stock_move_line.py

```diff
@@ -423,8 +445,16 @@ def _action_done(self):
                             # If a picking type is linked, we may have to create a production lot on
                             # the fly before assigning it to the move line if the user checked both
                             # `use_create_lots` and `use_existing_lots`.
-                            if ml.lot_name:
-                                ml_to_create_lot |= ml
+                            if ml.lot_name and not ml.lot_id:
```

Comment:
there will be an error anyway if the lot already exist. So this search  + warning is not necessary

Reply:
I think I understand why this code is never executed.
When you add a lot in the lot_ids field, it create the record in database on the fly. So lot_name and lot_id are set on the stock_move_line before you click on Validate.
Thus, I would remove this block of code.